src/ucp/api/ucp.h

```diff
@@ -2100,6 +2137,35 @@ ucs_status_t ucp_context_query(ucp_context_h context_p,
                                ucp_context_attr_t *attr);
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief Compare two remote keys according to parameters
+ *
+ * This routine returns equal if all the transports can compare equal and
+ * configurations are equal. At least one compare mode is needed.
+ *
+ * The comparison result is < 0 if rkey1 is lower than rkey2, 0 if they
+ * are equal or > 0 if rkey1 is greater than rkey2.
+ *
+ * If rkey2 contains one or more transports that are not comparable,
+ * appropriate error is returned to allow the application to not try to later
+ * deduplicate against it. The api can be called with rkey1 equal to NULL, rkey2
+ * is considered greater than the NULL key.
+ *
+ * @param [in]  context     Handle to @ref ucp_context_h
+ * @param [in]  params      Used to tune the type of comparison requested
+ * @param [in]  rkey1       First rkey to compare, can be null
```

Comment:
why we say it can be null?

Reply:
removed, as per https://github.com/openucx/ucx/pull/9289#discussion_r1297110413